leader-election-impl/src/main/java/com/palantir/paxos/SplittingPaxosStateLog.java

```
@@ -0,0 +1,81 @@
+ * determined by the cutoff point. If a read or write does occur prior to the cutoff point, i.e., to the legacy delegate
+ * e update the appropriate metric. Remaining methods are delegated only to the current delegate.
+ */
+public class SplittingPaxosStateLog<V extends Persistable & Versionable> implements PaxosStateLog<V> {
+    private final PaxosStateLog<V> legacyLog;
+    private final PaxosStateLog<V> currentLog;
+    private final Runnable markLegacyWrite;
+    private final Runnable markLegacyRead;
+    private final long cutoffInclusive;
+
+    public SplittingPaxosStateLog(PaxosStateLog<V> legacyLog,
+            PaxosStateLog<V> currentLog,
+            Runnable markLegacyWrite,
+            Runnable markLegacyRead,
+            long cutoffInclusive) {
+        this.legacyLog = legacyLog;
+        this.currentLog = currentLog;
+        this.markLegacyWrite = markLegacyWrite;
+        this.markLegacyRead = markLegacyRead;
+        this.cutoffInclusive = cutoffInclusive;
+    }
+
+    @Override
+    public void writeRound(long seq, V round) {
+        if (seq >= cutoffInclusive) {
+            currentLog.writeRound(seq, round);
+        } else {
+            markLegacyWrite.run();
+            legacyLog.writeRound(seq, round);
+        }
+    }
+
+    @Override
+    public byte[] readRound(long seq) throws IOException {
+        if (seq >= cutoffInclusive) {
+            return currentLog.readRound(seq);
+        } else {
+            markLegacyRead.run();
+            return legacyLog.readRound(seq);
+        }
+    }
+
+    @Override
+    public long getLeastLogEntry() {
+        return currentLog.getLeastLogEntry();
+    }
+
+    @Override
+    public long getGreatestLogEntry() {
+        return currentLog.getGreatestLogEntry();
+    }
+
+    @Override
+    public void truncate(long toDeleteInclusive) {
+        currentLog.truncate(toDeleteInclusive);
```

Seems like we should truncate both? Or just throw UnsupportedOperationException given where we're deploying this